avc/src/start_detector/start_detector.cpp

```diff
@@ -0,0 +1,67 @@
+#include "start_detector.h"
+#include <pluginlib/class_list_macros.h>
+
+using namespace std;
+using namespace ros;
+using namespace cv;
+
+namespace avc {
+    const Scalar red_low{0, 0, 100};
+    const Scalar red_high{100, 100, 255};
+
+    void start_detector::ImageCB(const sensor_msgs::ImageConstPtr &msg) {
+        cv_bridge::CvImageConstPtr cv_ptr;
+
+        try {
+            cv_ptr = cv_bridge::toCvShare(msg, "bgr8");
+        } catch (cv_bridge::Exception &e) {
+            ROS_ERROR("CV-Bridge error: %s", e.what());
+            return;
+        }
+        
+        const Mat &frameBGR = cv_ptr->image;
+        Mat blurredImage;
+        GaussianBlur(frameBGR, blurredImage, Size{7, 7}, 3);
+        Mat redMask = Mat::zeros(blurredImage.rows, blurredImage.cols, CV_8U);
+        inRange(blurredImage, red_low, red_high, redMask);
+        Mat redImage;
+        bitwise_and(blurredImage, blurredImage, redImage, redMask);
```

Comment:
This method definitely ANDs the image with itself and then applies the mask.
The desired behavior can be more cleanly achieved with this:
```
blurredImage.copyTo(redImage, redMask);
```